Orange/widgets/visualize/owfreeviz.py

```diff
@@ -0,0 +1,925 @@
+        pass
+
+    def commit(self):
+        selected = annotated = components = None
+        graph = self.graph
+        if self.data is not None and self.plotdata.validmask is not None:
+            name = self.data.name
+            metas = () + self.data.domain.metas + (self.variable_x, self.variable_y)
+            domain = Domain(attributes=self.data.domain.attributes,
+                            class_vars=self.data.domain.class_vars,
+                            metas=metas)
+            data = self.plotdata.data.transform(domain)
+            validmask = self.plotdata.validmask
+            mask = np.array(validmask, dtype=int)
+            mask[mask == 1] = graph.selection if graph.selection is not None \
+                else [False * len(mask)]
+            selection = np.array([], dtype=np.uint8) if mask is None else np.flatnonzero(mask)
+            if len(selection):
+                selected = data[selection]
+                selected.name = name + ": selected"
+                selected.attributes = self.data.attributes
+            if graph.selection is not None and np.max(graph.selection) > 1:
+                annotated = create_groups_table(data, mask)
+            else:
+                annotated = create_annotated_table(data, selection)
+            annotated.attributes = self.data.attributes
+            annotated.name = name + ": annotated"
+
+            comp_domain = Domain(
+                self.data.domain.attributes,
+                metas=[StringVariable(name='component')])
+
+            metas = np.array([["FreeViz 1"], ["FreeViz 2"]])
+            components = Table.from_numpy(
+                comp_domain,
+                X=self.plotdata.anchors.T,
+                metas=metas)
+
+            components.name = name + ": components"
+
+        self.Outputs.selected_data.send(selected)
+        self.Outputs.annotated_data.send(annotated)
+        self.Outputs.components.send(components)
+
+    def send_report(self):
+        if self.data is None:
+            return
+
+        def name(var):
+            return var and var.name
+
+        caption = report.render_items_vert((
+            ("Color", name(self.graph.attr_color)),
+            ("Label", name(self.graph.attr_label)),
+            ("Shape", name(self.graph.attr_shape)),
+            ("Size", name(self.graph.attr_size)),
+            ("Jittering", self.graph.jitter_size != 0 and "{} %".format(self.graph.jitter_size))))
+        self.report_plot()
+        if caption:
+            self.report_caption(caption)
+
+
+def add_circle(x, y):
+    CIRCLE_RADIUS = 0.033
+    func = lambda x, trigo: np.fromfunction(
+        lambda i: x + CIRCLE_RADIUS * trigo(2. * np.pi * i / 120.), (121,),
+        dtype=int)
+    circle_x = func(x, np.cos)
+    circle_y = func(y, np.sin)
+    color = QColor(0, 0, 255)  # blue
+    circle = pg.PlotCurveItem(
+        x=circle_x, y=circle_y,
+        pen=pg.mkPen(color, width=3),
+        antialias=False
```

Comment:
`antiAlias=True`

Reply:
However, I think that this looks better and more informative.

```
class MoveIndicator(pg.GraphicsObject):
    def __init__(self, x, y, parent=None, line=QLineF(), text="", **kwargs):
        super().__init__(parent, **kwargs)
        self.arrows = [
            pg.ArrowItem(pos=(x - 0.07 * cos(radians(angle)),
                              y + 0.07 * sin(radians(angle))),
                         parent=self, angle=angle,
                         headLen=13, tipAngle=45,
                         brush=pg.mkColor(128, 128, 128))
            for angle in (0, 90, 180, 270)]

    def paint(self, painter, option, widget):
        pass

    def boundingRect(self):
        return QRectF()


def add_circle(x, y):
    return MoveIndicator(x, y)
```

Of course, eliminate `add_circle` and use `MoveIndicator` directly in the code.

Reply:
Yes, that looks better.